app/src/main/java/com/futurumgame/base/serialization/parsing/BaseParseRule.java

```diff
@@ -10,28 +10,27 @@ protected BaseParseRule() {
         clearReadChars();
     }
 
-    public String getReadChars(){
+    public final String getReadChars(){
         return readChars;
     }
 
     @Override
     public ParseResult<T> next(char c) {
         readChars += c;
-        return ParseResult.create(null);
+        return ParseResult.failResult();
     }
 
     @Override
     public ParseResult<T> next(String string) {
         readChars += string;
-        return ParseResult.create(null);
+        return ParseResult.failResult();
     }
 
-    @Override
-    public String getParsingValue(T obj) {
-        return null;
+    public final void clearReadChars() {
+        clear();
+        readChars = "";
     }
 
-    protected void clearReadChars() {
-        readChars = "";
+    protected void clear() {
```

Comment:
Why is this method completely empty? Delete it when it does nothing

Reply:
virtual method that can be overriden in case there is more to clear, renamed the method to make things more clear